exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/DrillSqlWorker.java

```diff
@@ -116,20 +117,32 @@ public DrillSqlWorker(QueryContext context) {
 
   private RuleSet[] getRules(QueryContext context) {
     StoragePluginRegistry storagePluginRegistry = context.getStorage();
-    RuleSet drillLogicalRules = DrillRuleSets.mergedRuleSets(
+
+    // Ruleset for the case where VolcanoPlanner is used for everything : join, filter/project pushdown, partition pruning.
+    RuleSet drillLogicalVolOnlyRules = DrillRuleSets.mergedRuleSets(
         DrillRuleSets.getDrillBasicRules(context),
+        DrillRuleSets.getPruneScanRules(context),
         DrillRuleSets.getJoinPermRules(context),
         DrillRuleSets.getDrillUserConfigurableLogicalRules(context));
+
+    // Ruleset for the case where join planning is done in Hep-LOPT, filter/project pushdown and parttion pruning are done in VolcanoPlanner
+    RuleSet drillLogicalHepJoinRules = DrillRuleSets.mergedRuleSets(
+        DrillRuleSets.getDrillBasicRules(context),
+        DrillRuleSets.getPruneScanRules(context),
+        DrillRuleSets.getDrillUserConfigurableLogicalRules(context));
+
+    // Ruleset for the case where join planning and partition pruning is done in Hep, filter/project pushdown are done in VolcanoPlanner
```

Comment:
Exactly. The ruleset defined here is used in Frameworks.VolcanoPlanner. In case that join / partition pruning is done in Hep, we do not put the join/partition pruning rules in the ruleset here. In stead, the join/partition pruning rules are specified when HepPlanner is built.